### PR TITLE
Learner-5298 Coupon report fixed for course entitlement

### PR DESCRIPTION
**Pull request by UsamaSadiq**
LEARNER-5298

Coupon report generation uses course_key attribute to generate the report.
Course entitlement does not have a course_key attribute which was causing
the issue so instead used entitlement uuid.

### CODE REVIEW

File: ecommerce/extensions/voucher/tests/test_utils.py [hunk 1]
```diff
@@ -27,10 +27,12 @@
     generate_coupon_report,
     get_voucher_and_products_from_code,
     get_voucher_discount_info,
-    update_voucher_offer
+    update_voucher_offer,
 )
 from ecommerce.tests.mixins import LmsApiMockMixin
 from ecommerce.tests.testcases import TestCase
+from ecommerce.entitlements.utils import create_or_update_course_entitlement
+
```
Comment (uzairr) [inline ecommerce/extensions/voucher/tests/test_utils.py]: Only one line is required please remove the extra one.

**Review by uzairr — APPROVED** [review]

File: ecommerce/extensions/voucher/utils.py [hunk 2]
```diff
@@ -100,10 +99,29 @@ def _get_discount_info(discount_data):
...
+        course_id (str or None if entitlement)
+        course_organization (str or None if entitlement)
+    """
+    course_organization = None
+    if seat_stockrecord.product.is_course_entitlement_product:
+        course_id = None
+    else:
+        course_id = seat_stockrecord.product.attr.course_key
+        course_organization = CourseKey.from_string(course_id).org
+    return course_id, course_organization
```
Comment (cpennington) [inline ecommerce/extensions/voucher/utils.py:119]: Nit: Would this be clearer written as?

    if seat_stockrecord.product.is_course_entitlement_product:
        return (None, None)
    else:
        course_id = seat_stockrecord.product.attr.course_key
        course_organization = CourseKey.from_string(course_id).org
        return course_id, course_organization

File: ecommerce/extensions/voucher/tests/test_utils.py [hunk 3]
```diff
@@ -62,13 +63,11 @@ def setUp(self):
 
         self.course = CourseFactory(id='course-v1:test-org+course+run', partner=self.partner)
         self.verified_seat = self.course.create_or_update_seat('verified', False, 100)
-
```
Comment (waheedahmed) [inline ecommerce/extensions/voucher/tests/test_utils.py]: Add this and all other removed empty lines back.

File: ecommerce/extensions/voucher/tests/test_utils.py [hunk 4]
```diff
@@ -167,8 +182,12 @@ def use_voucher(self, order_num, voucher, user):
             users (list): list of users
         """
         order = OrderFactory(number=order_num)
+        if add_entitlement:
+            order_line = OrderLineFactory(product=self.entitlement)
+            order.lines.add(order_line)
         order_line = OrderLineFactory(product=self.verified_seat)
         order.lines.add(order_line)
+
```
Comment (waheedahmed) [inline ecommerce/extensions/voucher/tests/test_utils.py]: Remove this unnecessary new line.

**Review by waheedahmed — APPROVED** [review]
Some minor nits, 👍 once these and Cale's comment resolved.

**Comment by UsamaSadiq** [comment]
@cpennington thanks for the suggestion. It sure looks more neat the way you suggested so I have changed the code according to your suggestion.

File: ecommerce/extensions/voucher/tests/test_utils.py (hunk 3 above)
Comment (UsamaSadiq) [reply ecommerce/extensions/voucher/tests/test_utils.py]: Updated the files as per your suggestion.

File: ecommerce/extensions/voucher/tests/test_utils.py (hunk 4 above)
Comment (UsamaSadiq) [reply ecommerce/extensions/voucher/tests/test_utils.py]: Removed extra lines.

**Pull request merged**